test/active_model_serializers/register_jsonapi_renderer_test_isolated.rb

```diff
@@ -12,7 +12,9 @@ class << self
     end
 
     def render_with_jsonapi_renderer
-      author = Author.new(params[:data][:attributes])
+      unlocked_params = Rails::VERSION::MAJOR >= 5 ? params.to_unsafe_h : params
+      attributes = unlocked_params[:data].present? ? unlocked_params[:data][:attributes] : {}
```

Comment:
So, I gather `Author.new(params[:data][:attributes])` is failing for some reason. perhaps we should instead `author = Author.new(params.require(:data).require(:attributes))`?  I'm not sure what the need is to cast the ActionController::Parameters or when `data` wouldn't be present. 

In actual usage I would use the JsonApi deserializer here.